internal/staging/stdaddress/README.md

```diff
@@ -0,0 +1,195 @@
+stdaddress
+==========
+
+[![Build Status](https://github.com/decred/dcrd/workflows/Build%20and%20Test/badge.svg)](https://github.com/decred/dcrd/actions)
+[![ISC License](https://img.shields.io/badge/license-ISC-blue.svg)](http://copyfree.org)
+[![Doc](https://img.shields.io/badge/doc-reference-blue.svg)](https://pkg.go.dev/github.com/decred/dcrd/internal/staging/stdaddress)
+
+## Decred Address Overview
+
+An address is a human-readable representation of a possible destination for a
+payment.  More specifically, it encodes information needed to create a smart
+contract via a transaction script that imposes constraints required to redeem
+funds.
+
+Under the hood, an address typically consists of 4 primary components:
+
+- The network for which the address is valid
+- The data necessary to create a payment script with the appropriate constraints
+- The script version of the aforementioned script
+- A checksum to prevent errors due to improper entry
+
+It is also worth keeping in mind that the script version may be implicit
+depending on the specific concrete encoding of the address.
+
+### Supported Version 0 Addresses
+
+The following table lists the `version 0` address types this package supports
+along with whether the type is supported by the staking system, whether it has
+an associated `hash160`, and some additional notes and recommendations:
+
+Version 0 Address Type    | Staking? | Hash160? | Notes / Recommendations
+--------------------------|----------|----------|----------------------------------
+p2pk-ecdsa-secp256k1      |    N     |    N     | Prefer p2pkh in on-chain txns [1]
+p2pk-ed25519              |    N     |    N     | Not recommended [2]
+p2pk-schnorr-secp256k1    |    N     |    N     | Prefer p2pkh, single party [1,3]
+**p2pkh-ecdsa-secp256k1** |  **Y**   |  **Y**   | **Preferred v0 address**
+p2pkh-ed25519             |    N     |    Y     | Not recommended [2]
+p2pkh-schnorr-secp256k1   |    N     |    Y     | Only use with single party [3]
+p2sh                      |    Y     |    Y     | -
+
+Abbreviations:
+
+* p2pk = Pay-to-public-key
+* p2pkh = Pay-to-public-key-hash
+* p2sh = Pay-to-script-hash
+
+Notes:
+
+- [1] Pay to public key addresses are only recommended for use in sharing public
+      keys via off-chain protocols such as multi-party signature creation.  Pay
+      to public key hash addresses should be used for on-chain transactions to
+      avoid revealing the public key until absolutely necessary.
```

Comment:
Should this be `unless absolutely necessary`?

Reply:
I used until because it eventually has to be revealed to redeem the funds.  Unless implies, to me at least, that it doesn't necessarily have to ever be revealed which is not the case.